### ISSUE TITLE
Byte range support fails on noaa-goes16 S3 Bucket

### ISSUE THREAD

**Issue by dopplershift** [Unidata/netcdf-c#2642]
As noted on conda-forge/libnetcdf-feedstock#166, the following URL https://noaa-goes16.s3.amazonaws.com/ABI-L1b-RadC/2017/059/03/OR_ABI-L1b-RadC-M3C13_G16_s20170590337505_e20170590340289_c20170590340316.nc#mode=bytes fails when trying to use `ncdump -h` or netcdf4-python's `Dataset` class with: 

```
ncdump: https://noaa-goes16.s3.amazonaws.com/ABI-L1b-RadC/2017/059/03/OR_ABI-L1b-RadC-M3C13_G16_s20170590337505_e20170590340289_c20170590340316.nc#mode=bytes: NetCDF: Malformed URL
```

Adjusting it to https://noaa-goes16.s3.us-east-1.amazonaws.com/ABI-L1b-RadC/2017/059/03/OR_ABI-L1b-RadC-M3C13_G16_s20170590337505_e20170590340289_c20170590340316.nc#mode=bytes fixes the problem, but this should not be necessary based on the [AWS S3 docs](https://docs.aws.amazon.com/general/latest/gr/s3.html) on supported endpoints for the us-east-1 region.

I'll also note that the URL in question comes from netcdf-c's own `tst_byterange.c`: https://github.com/Unidata/netcdf-c/blob/00a722b253bae186bba403d0f92ff1eba719591f/nc_test/tst_byterange.c#L32-L41

Before any work is done to correct the URL parsing/checking, that test has to be investigated to identify why it's not already failing.

**Comment by DennisHeimbigner** [comment]
Fixed by PR https://github.com/Unidata/netcdf-c/pull/2649

**Comment by zklaus** [comment]
Should this and its sister issue #2643 be closed following #2649?
